dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -327,4 +330,32 @@ object TypedEncoder {
     recordEncoder: Lazy[RecordEncoderFields[G]],
     classTag: ClassTag[F]
   ): TypedEncoder[F] = new RecordEncoder[F, G]
+
+  type UDT[A >: Null] = FramelessInternals.PublicUserDefinedType[A]
+
+  /**
+    * Encodes things using a Spark SQL's User Defined Type (UDT) if there is one defined in implicit.
+    *
+    * Example: to use Spark ML's VectorUDT implementation as a TypedEncoder, add the following in your scope:
+    * {{{
+    * import org.apache.spark.ml.linalg.{Vector, SQLDataTypes}
+    * import frameless.TypedEncoder.UDT
+    * implicit val mLVectorUDT: UDT[Vector] = SQLDataTypes.VectorType.asInstanceOf[UDT[Vector]]
```

Comment:
Hm.. probably it isn't a bit problem as soon as we declare package with implicits for UDTs in spark-ml.